examples/ota-provider-app/linux/main.cpp

```diff
@@ -407,9 +407,31 @@ void ApplicationInit()
 
 void ApplicationShutdown() {}
 
+namespace {
+class OtaProviderAppMainLoopImplementation : public AppMainLoopImplementation
+{
+public:
+    void RunMainLoop() override { chip::DeviceLayer::PlatformMgr().RunEventLoop(); }
+    void SignalSafeStopMainLoop() override
+    {
+        chip::DeviceLayer::PlatformMgr().ScheduleWork([](intptr_t) {
+            ChipLogDetail(SoftwareUpdate, "Scheduling BdxOtaSender to ABORT TRANSFER");
+
+            gOtaProvider.GetBdxOtaSender()->AbortTransfer();
+        });
+
+        chip::Server::GetInstance().GenerateShutDownEvent();
+
+        chip::DeviceLayer::PlatformMgr().ScheduleWork([](intptr_t) { chip::DeviceLayer::PlatformMgr().StopEventLoopTask(); });
```

Comment:
Why the split? Schedule work will run async, so this would rely on sequencing. Could we combine the two schedulework?